rc/extra/ranger.kak

```diff
@@ -14,30 +14,30 @@ define-command \
   -docstring %{ranger [<arguments>]: open the file system explorer to select buffers to open
 All the optional arguments are forwarded to the ranger utility} \
   ranger %{ %sh{
-  if [ -n "$TMUX" ]; then
+  if [ -n "${TMUX}" ]; then
     tmux split-window -h \
-      ranger $@ --cmd " \
+      ranger "$@" --cmd " \
         map <return> eval \
           fm.execute_console('shell \
-            echo evaluate-commands -client $kak_client edit {file} | \
-            kak -p $kak_session; \
+            printf %s evaluate-commands -client ' + shell_escape('$kak_client') + ' edit {file} | \
+            kak -p ' + shell_escape('$kak_session') + '; \
             tmux select-pane -t $kak_client_env_TMUX_PANE'.format(file=fm.thisfile.path)) \
           if fm.thisfile.is_file else fm.execute_console('move right=1')"
-  elif [ -n "$kak_client_env_STY" ]; then
+  elif [ -n "${STY}" ]; then
     tty="$(ps -o tty ${kak_client_pid} | tail -n 1)"
     args="${@:-$PWD}"
     screen -X eval \
         'split -h' \
         'focus down' \
-        "screen ranger $args
+        "screen ranger \"$args\"
```

Comment:
Actually maybe this shouldn't be escaped, as the variable will contain command line parameters?

Reply:
I was quoting this in order to ensure that the current directory (e.g. with spaces in them) was passed correctly.  I've since changed this to use the screen 'chdir` command.

As you've noted, quoting causes command line arguments to fail.  Indeed, I hadn't been testing for this correctly.

However, even unquoted, the problem remains that passing command line arguments to ranger doesn't work.  Consider, in kak: `ranger --cmd 'cd /etc'`
Using the unquoted forumula, screen seems to separate the command its evaluating at `--cmd` and tries to evaluate `cd` as a screen command.

I don't yet have a solution.

Reply:
@lenormf: Does passing command line arguments to ranger work in tmux or x11? I can't get it to work with the `ranger.kak` in master. My recent commit uses  `--choosefiles` and a heredoc shell script instead of `map`.  It's the only way I could get everything to work with screen.  If passing ranger command line args doesn't work in tmux or x11 perhaps we should switch them to `--choosefiles` and a heredoc shell script as well?

Reply:
You could wrap every argument in quotes, and pass this escaped string to `screen`, something like:

```bash
args=""
for i in "$@"; do
     args="${args} \"${i}\""
done

screen -X eval … "screen ranger ${args}…
```